### ISSUE TITLE
Dependency tag ref that contains no pawn.json results in ensure/resolve failure

### ISSUE THREAD

**Issue by Southclaws** [Southclaws/sampctl#137]
The point in time that tag 2.9.2 (and earlier) points to in the Streamer plugin was before sampctl/pawn.json was introduced to the repo. So when this dependency is cloned, there is no package definition present which results in a failure.

**Comment by Sunehildeep** [comment]
This one also doesn't have pawn.json but its being used in ScavengeSurvive, thought to check it and found this:
https://github.com/oscar-broman/strlib

**Comment by Southclaws** [comment]
When I worked on 1.7.12 I realised the cause of the issue wasn't this, and I thought about this a lot more.

If a project's dependencies change over time, then injecting the latest dependencies may be a bad idea - for example YSI 4.x and 5.x branches, if this was done using tags and 4.x had no definition file, injecting the latest one upon clone would result in unnecessary dependencies being pulled.

I'm still on the fence about this now, as implementing it would be quite a bit of work and hacky workarounds (mainly because the dependencies are actually git repositories and creating a fake pawn.json would cause more hassle when trying to run `ensure`).

The previous issue with the streamer is now resolved, the problem itself should never have caused a command failure, just a warning print. I think if users intend to use older libraries, they should just add the sub-package dependencies to their pawn.json manually instead of pulling possibly irrelevant or mismatching dependencies.
